### ISSUE TITLE
[React Native Gallery]: Landmarks are not defined in the React Native Gallery

### ISSUE THREAD

**Issue by protikbiswas100** [microsoft/react-native-gallery#716]
https://microsoft.visualstudio.com/OS/_workitems/edit/59420397/?view=edit

Build Details:
OS Version: 24H2 
OS Build: 26100.6584
App: React Native Gallery Preview
App Version: 1. 0.18.0
AT Used: Narrator

Repro-Steps:
Launch the React Native Gallery-Preview app.
Navigate to the 'Expand menu' button and invoke it.
Try to navigate landmarks using 'D/Shift D' or use 'CapsLock + Fn + F5' to see the list of landmarks.
Actual Result:
When using the shortcut 'D/Shift D' or use 'CapsLock + Fn + F5' to see the list of landmarks (Screen Reader landmark navigation), no landmarks are detected in the React Native Gallery. The user cannot navigate to main content or other landmark regions because no ARIA landmarks are defined.

Expected Result:
Landmarks such as main, banner, and Navigation should be defined. Screen reader users should be able to navigate between these landmarks using 'D/Shift D' and list should be accessible by using CapsLock + Fn + F5 without any issues.

User Impact:
Screen reader users rely on landmark navigation to quickly move between major sections of the application. Without defined landmarks, users must navigate sequentially through all elements, which significantly increases interaction time and reduces efficiency.

MAS Reference Link: [MAS 1.3.1 – Info and Relationships](https://aka.ms/MAS1.3.1)

Trap ID: [1.6 - Poor Grouping](https://aka.ms/PoorGrouping)

“Have feedback on bugs logged, please tag bug as “A11yRCA” and add your feedback in the comment section.”

**Comment by protikbiswas100** [comment]
https://github.com/microsoft/react-native-gallery/pull/721
